openwisp_controller/config/static/config/js/widget.js

```diff
@@ -153,6 +153,50 @@
         return error;
     };
 
+    var getAllVariables = function () {
+        var systemDefinedVariables = JSON.parse($('#system_context').html()),
+            configVariables;
+        if (window.getContext()) {
+            configVariables = JSON.parse(window.getContext().value);
+        }
+        var allVariables = Object.assign({}, systemDefinedVariables, configVariables);
+        return allVariables;
+    };
+
+    const debounce = (delay, func) => {
+        var debounceTimer;
+        return function() {
+            clearTimeout(debounceTimer);
+            debounceTimer = setTimeout(() => func.apply(this, arguments), delay);
+        };
+    };
+
+    var validateMaxLength = function() {
+        var allVariables = getAllVariables(),
+            variablesKeys = Object.keys(allVariables),
+            allFields = $(".jsoneditor .vTextField");
```

Comment:
We should try to cut down the number of fields we loop over.
At first page load, you could add a specific class to all the fields which have the maxlength attribute.
Then here you could filter out only those fields.